libs/wingsdk/src/shared-gcp/counter.inflight.ts

```diff
@@ -0,0 +1,60 @@
+import { Datastore } from "@google-cloud/datastore";
+import type { ICounterClient } from "../cloud";
+
+// const AMOUNT_TOKEN = "amount";
+// const INITIAL_VALUE_TOKEN = "initial";
+const COUNTER_ID = "counter";
+// const VALUE_ATTRIBUTE = "counter_value";
+// const SET_VALUE = "set_value";
+
+export class CounterClient implements ICounterClient {
+  private readonly client: Datastore;
+
+  constructor(
+    private readonly databaseName: string,
+    private readonly initial: number = 0
+  ) {
+    this.client = new Datastore({ databaseId: this.databaseName });
+  }
+
+  public async inc(amount = 1, key = COUNTER_ID): Promise<number> {
+    const kind = "Counter";
+    const name = key;
+    const counterKey = this.client.key([kind, name]);
+
+    // Fetch the current counter
+    const [existingCounter] = await this.client.get(counterKey);
+
+    // Determine the current value of the counter or use the initial value
+    const currentValue = existingCounter?.data?.count || this.initial;
```

Comment:
```suggestion
    const currentValue = existingCounter?.data?.count ?? this.initial;
```
if the current value is -1 and the initial is 6, 
after one inc- we'll go to: current 0 initial 6
after two inc- we'll go to: current 7,
can you notice why?